src/store/wizardSlice.ts

```diff
@@ -40,6 +42,13 @@ type wizardState = {
     shareMethod: AwsShareMethod;
     source: V1ListSourceResponseItem | undefined;
   };
+  azure: {
+    shareMethod: AzureShareMethod;
+    tenantId: string;
+    subscriptionId: string;
+    source: AzureUploadRequestOptions['source_id'] | undefined;
+    resourceGroup: AzureUploadRequestOptions['resource_group'];
```

Comment:
In this case, if you look at the type of `AzureUploadRequestOptions['source_id']`, it is just `string`. So this would theoretically be okay, but for consistency sake with what I’m suggesting above let’s change this type to just be `string`.

Reply:
just wander why the source in aws define as aws: {
    source: V1ListSourceResponseItem | undefined;
  };

Reply:
Oh, that's a good point, I forgot about that. You are right, for AWS we store the entire `V1ListSourceResponseItem` object, not just the id. 

Let's do the same thing here in order to be consistent. So in that case, the type here would not be `string` like I suggested, it would be `V1ListSourceResponseItem | undefined` as it is for AWS.

Reply:
not sure how you want to handel that for Azure, define only the source or also the resourceGroup?
if I define only the source, I think Iits require more changes in other places as well

Reply:
I think its more clear to define source and resourceGroup as well

Reply:
I tried to change the source to source: V1ListSourceResponseItem | undefined;
but its seems that there are some issues. not sure if that is the correct way for azure

Reply:
It works, so we can use this for now and worry about the small inconsistency with the AWS step later.